web-wallet/src/lib/containers/BridgeContract/BridgeContract.svelte

```diff
@@ -0,0 +1,87 @@
+<svelte:options immutable={true} />
+
+<script>
+  import { onDestroy, onMount } from "svelte";
+  import { browser } from "$app/environment";
+  import { collect, getKey, pick } from "lamb";
+  import { getBalance, watchBlocks } from "@wagmi/core";
+
+  import { Button } from "$lib/dusk/components";
+  import { account, modal, wagmiConfig } from "$lib/web3/walletConnection";
+  import { Bridge } from "$lib/components";
+  import { createCurrencyFormatter } from "$lib/dusk/currency";
+  import {
+    gasStore,
+    networkStore,
+    settingsStore,
+    walletStore,
+  } from "$lib/stores";
+
+  const { networkName } = $networkStore;
+  const collectSettings = collect([
+    pick(["gasLimit", "gasPrice"]),
+    getKey("language"),
+  ]);
+
+  /** @type {(() => void) | null} */
+  let unwatch = null;
+
+  /**
+   * @typedef { import("@wagmi/core").GetBalanceReturnType } GetBalanceReturnType
+   */
+  /** @type {GetBalanceReturnType | undefined}  */
+  let evmDuskBalance;
+
+  onMount(async () => {
+    if (!browser) return;
+
+    unwatch = watchBlocks(wagmiConfig, {
+      async onBlock() {
+        if ($account.isConnected && $account.chainId && $account.address) {
+          try {
+            evmDuskBalance = await getBalance(wagmiConfig, {
+              address: $account.address,
+              chainId: $account.chainId,
+            });
+          } catch (e) {
+            // eslint-disable-next-line no-console
+            console.error("getBalance failed", e);
+            evmDuskBalance = undefined;
+          }
+        }
+      },
+    });
+  });
+
+  onDestroy(() => {
+    if (unwatch) {
+      unwatch();
+      unwatch = null;
+    }
+  });
+
+  $: ({ isConnected } = $account);
+  $: ({ balance, currentProfile } = $walletStore);
+  $: [gasSettings, language] = collectSettings($settingsStore);
+  $: gasLimits = $gasStore;
+  $: unshieldedAddress = currentProfile
+    ? currentProfile.account.toString()
+    : "";
+  $: duskFormatter = createCurrencyFormatter(language, "DUSK", 9);
+</script>
+
+{#if ["mainnet", "testnet", "devnet", "localnet"].includes(networkName)}
```

Comment:
Are you sure this check belongs here?
I think it should be responsibility of the parent to check if this contract can be used and warn the user otherwise.